sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobLegalHoldResult.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.models;
+
+/**
+ * The blob legal hold result.
+ */
+public class BlobLegalHoldResult {
+
+    private final boolean hasLegalHold;
+
+    /**
+     * Creates a new BlobLegalHoldResult
+     * @param hasLegalHold whether or not a legal hold is enabled on the blob.
+     */
+    public BlobLegalHoldResult(boolean hasLegalHold) {
```

Comment:
I wonder if we should convert this class to interface and serve model from "impl" package. This would avoid constructor explosion if more properties come.
We should start doing this for immutable models.